src/collection-browser.ts

```diff
@@ -1498,7 +1513,9 @@ export class CollectionBrowser
    * Call this whenever the counts are invalidated (e.g., by a query change).
    */
   private refreshLetterCounts(): void {
-    this.prefixFilterCountMap = {};
+    if (Object.keys(this.prefixFilterCountMap).length > 0) {
+      this.prefixFilterCountMap = {};
+    }
```

Comment:
Just preventing needless component update cycles, by ensuring we only replace this reactive property if it wasn't already empty.